components/tidb_query/src/batch/executors/table_scan_executor.rs

```diff
@@ -18,6 +19,8 @@ use crate::Result;
 
 pub struct BatchTableScanExecutor<S: Storage>(ScanExecutor<S, TableScanExecutorImpl>);
 
+type HandleIndicesVec = SmallVec<[usize; 2]>;
```

Comment:
What does the number `2` come from? Does TiDB pass at most 2 PK handles now?

Reply:
It means if there are no more than 2 items in the vec, it will alloc the array on stack; otherwise on heap, like how Vec does. PKs passed from TiDB is excepted to be few, accroding to @breeswish , so I think it's ok to set it 2.